packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -68,6 +69,7 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
         };
 
         const logger2 = ChildLogger.create(logger, "OdspDriver");
+        const fetcher = new Fetcher(undefined, logger2);
```

Comment:
is there a reason to create it here?
OdspDocumentService.create (called form createDocumentService()) is already fine with creating it.

Reply:
Because we need it when creating the file, to initialize it with epoch. In non create new case, it will created from there.

Reply:
I'm not sure I follow. I think all code paths are the same - the very first call to SPO (whether it create a file, or get snapshot for a file, or something else) populates epoch. Until then it's undefined.
In other words, this logic is fully within Fetcher instance, driver instance just need to make sure it always creates Fetcher right away.

Reply:
The file is create in factory not in driver service. So in that case, it needs to created in factory and then passed to createDocumentService. So I think in both cases, I will create it in factory.